modules/markets-index/src/main/scala/org/ergoplatform/dex/index/processes/PoolsIndexing.scala

```diff
@@ -62,12 +73,21 @@ object PoolsIndexing {
         val assets        = poolSnapshots.flatMap(p => List(p.entity.lp.id, p.entity.x.id, p.entity.y.id)).distinct
         val commit        = batch.lastOption.map(_.commit)
 
+        val retryPolicy = limitRetries[F](conf.limitRetries) |+| constantDelay[F](conf.retryDelay)
         val resolveNewAssets =
           for {
             existingAssets <-
               txr.trans(NonEmptyList.fromList(assets).fold(List.empty[TokenId].pure[D])(repos.assets.existing))
             unknownAssets = assets.diff(existingAssets)
-            assetsInfo <- unknownAssets.filterNot(_ == ErgoAssetId).traverse(explorer.getTokenInfo)
+            assetsInfo <- unknownAssets
+                            .filterNot(_ == ErgoAssetId)
+                            .flatTraverse(tknId =>
+                              retryingOnFailures(
+                                retryPolicy,
+                                (opt: Option[TokenInfo]) => opt.isDefined,
+                                (_: Option[TokenInfo], _: RetryDetails) => info"Failed to find token $tknId. Retrying..."
+                              )(explorer.getTokenInfo(tknId)).map(_.toList)
```

Comment:
Retry makes sense only in the case of non-2xx/4xx response code. In the case of 4xx code indexer should panic.